scripts/coe_stats_gui.lua

```diff
@@ -0,0 +1,90 @@
+---@class coe.StatsGUI
+local StatsGUI = {}
+
+local mod_gui = require( "mod-gui" )
+
+-- =============================================================================
+
+local function build_stats_info_frame( statistics_frame )
+  local info_frame = statistics_frame.add{
+    type = "frame",
+    name = "coe_stats_info_frame",
+    direction = "vertical"
+  }
+
+  info_frame.add{
+    type = "label",
+    caption = {"coe-stats-gui.label_launches_title"}
+  }.parent.add{
+    type = "flow",
+    direction = "horizontal"
+  }.add{
+    type = "label",
+    caption = {"coe-stats-gui.label_launches_required"}
+  }.parent.add{
+    type = "label",
+    caption = tostring( global.rocket_silo.required_launches )
+  }.parent.parent.add{
+    type = "flow",
+    direction = "horizontal"
+  }.add{
+    type = "label",
+    caption = {"coe-stats-gui.label_launches_complete"}
+  }.parent.add{
+    type = "label",
+    caption = tostring( global.rocket_silo.total_launches )
+  }
+
+  return info_frame
+end -- build_stats_info_frame
+
+-------------------------------------------------------------------------------
+
+local function open_stats_ui(player)
+  local gui = player.gui.left
+  local statistics_frame =
+      gui.add {
+      type = "frame",
+      name = "coe_statistics_frame",
+      direction = "vertical",
+      caption = {"coe-stats-gui.label_stats_title"}
+  }
+
+  build_stats_info_frame( statistics_frame )
+end -- open_stats_ui
+
+-------------------------------------------------------------------------------
+
+function StatsGUI.onPlayerCreated( event )
+  local player = game.get_player(event.player_index)
+  local dialog = mod_gui.get_button_flow( player )
+  if dialog.coe_button_statistics then dialog.coe_button_statistics.destroy() end
```

Comment:
This check isn't needed

Reply:
I tried it without it and got an error on first run.

Reply:
Was this on a new map? hmmm, Given that onPlayerCreated is a rare event the check here doesn't hurt anything.